### ISSUE TITLE
[Bug] Expander within CollectionView does not work properly on iOS

### ISSUE THREAD

**Issue by sachinkanadia** [xamarin/XamarinCommunityToolkit#608]
### Description
When using the XF expander view within the XF collection view the expanding action performs in irregular fashion. All the expenders below the current expander that has been tapped expand. The only expander that should expand is the one that has been tapped. This only occurs on iOS. Android works fine.

### Steps to Reproduce

1. Download the sample code provided in the re-production link.
2. Run the sample on an iOS device or simulator.
3. Tap on an expander within the collection.

### Expected Behavior
Only the expander that has been tapped should expand.
### Actual Behavior
Other expanders expand too.
### Basic Information

- Version with issue: XF 4.8.0.1687
- Last known good version: N/A
- Platform Target Frameworks:
  - iOS: 14.2

### Environment

```
=== Visual Studio Community 2019 for Mac ===

Version 8.8.1 (build 37)
Installation UUID: c6ad9608-c31d-49fa-acff-5f1a74f47eef
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000093

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.93 (2020-02/620cf538206) (64-bit)
	Package version: 612000093

=== Roslyn (Language Service) ===

3.8.0-5.20515.7+713efb0e68d34ab75faaf709e071e5c2d8d4ed0e

=== NuGet ===

Version: 5.8.0.6860

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.100/Sdks
SDK Versions:
	5.0.100
	3.1.403
	3.1.402
	3.1.401
	3.1.300
	3.1.200
	3.1.102
	3.1.100
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.0
	3.1.9
	3.1.8
	3.1.7
	3.1.4
	3.1.2
	3.1.0
	2.1.23
	2.1.22
	2.1.21
	2.1.18
	2.1.17
	2.1.16
	2.1.15
	2.1.14

=== .NET Core 3.1 SDK ===

SDK: 3.1.403

=== Xamarin.Profiler ===

Version: 1.6.13.11
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin Designer ===

Version: 16.8.0.507
Hash: e87b24884
Branch: remotes/origin/d16-8
Build date: 2020-10-29 00:31:38 UTC

=== Apple Developer Tools ===

Xcode 12.2 (17535)
Build 12B45b

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 14.6.0.15 (Visual Studio Community)
Hash: 87a1b18d8
Branch: d16-8
Build date: 2020-11-16 21:39:41-0500

=== Xamarin.Android ===

Version: 11.1.0.17 (Visual Studio Community)
Commit: xamarin-android/d16-8/c0e2b8e
Android SDK: /Users/sachin/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.5
SDK Build Tools Version: 29.0.2

Build Information: 
Mono: be2226b
Java.Interop: xamarin/java.interop/d16-8@79d9533
ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
SQLite: xamarin/sqlite/3.32.1@1a3276b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/sachin/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.8.0.32
Hash: 01a7774
Branch: remotes/origin/d16-8
Build date: 2020-10-13 23:32:30 UTC

=== Android Device Manager ===

Version: 16.8.0.45
Hash: fc0af5f
Branch: remotes/origin/d16-8
Build date: 2020-10-13 23:32:54 UTC

=== Build Information ===

Release ID: 808010037
Git revision: acd8ddc086768c701bd641d6ec27316f45813175
Build date: 2020-11-17 12:49:41-05
Build branch: release-8.8
Xamarin extensions: acd8ddc086768c701bd641d6ec27316f45813175

=== Operating System ===

Mac OS X 10.16.0
Darwin 20.1.0 Darwin Kernel Version 20.1.0
    Sat Oct 31 00:07:11 PDT 2020
    root:xnu-7195.50.7~2/RELEASE_X86_64 x86_64

```

### Reproduction Link

https://github.com/sachinkanadia/ExpanderSample

**Comment by sachinkanadia** [comment]
Is there any update on this?

**Comment by sachinkanadia** [comment]
Not sure why this has been allocated to the XamarinCommunityToolkit project. Surely this is an issue with the CollectionView rather than the expander control?

**Comment by pictos** [comment]
@AndreiMisiukevich, can you please confirm if we can solve this issue here?

**Comment by AndreiMisiukevich** [comment]
> @AndreiMisiukevich, can you please confirm if we can solve this issue here?

No, unfortunately, I don't think so.
I am pretty sure that now Expander **does not** work with List/Collection Views.

**Comment by sachinkanadia** [comment]
Am I right in saying that in order to resolve this issue one would need to update the collection view control code rather than the expander control code?

**Comment by AndreiMisiukevich** [comment]
I think, yes. You are right

**Comment by sachinkanadia** [comment]
Thank you @AndreiMisiukevich 
Can we move this to the correct project please. It should be allocated to the project responsible for the XF Collection View Control.

**Comment by pictos** [comment]
@AndreiMisiukevich what do you think (I don't if it's feasible) to add a Warning when the parent is a CollectionView or ListView? That way if the dev doesn't read the docs, the output will alert him.

**Comment by sachinkanadia** [comment]
@pictos I don't feel adding a warning to resolve this defect is adequate. We need to move this defect to the correct place in order to allow the Xamarin Forms team to respond. Please can this be arranged.

**Comment by pictos** [comment]
@sachinkanadia I didn't say that this is the solution to the bug. We agreed that this bug should be on the XF repo and be solved there (until now) on our side, what we can do until they fix this bug is warning devs that isn't a good idea to use Expander inside collections. By the way, can you open this issue there?

**Comment by sachinkanadia** [comment]
@pictos my bad. Is there a way we can transfer this issue there? It's just that if I raise another bug there, theres always the chance it can be moved elsewhere again. I agree using a warning in the meantime would be a good idea.

**Comment by pictos** [comment]
@sachinkanadia don't worry, English isn't my main language, so I probably didn't express myself very well. Ok. @jfversluis @jsuarezruiz can you move this issue to the XF repo? Thanks in advance.

**Comment by jfversluis** [comment]
Before moving this (again) I would like to 100% determine this is a CollectionView issue and not an Expander issue. How we should do that I don't know, but I'm not a fan of trying to "blame" other people without having data to back it up.

@AndreiMisiukevich what makes you think it's in the CollectionView? I don't know how the expand action works, but it almost sounds like something is triggered for all Expanders (on iOS). Can we make a Expander unique in some way? Or is that something that needs to happen in the CollectionView?

**Comment by AndreiMisiukevich** [comment]
> @AndreiMisiukevich what makes you think it's in the CollectionView? I don't know how the expand action works, but it almost sounds like something is triggered for all Expanders (on iOS). Can we make a Expander unique in some way? Or is that something that needs to happen in the CollectionView?

1) Expander works fine with Scroll + Stack
2) Expander works fine for CollectionView on Android (if I am not wrong)

**Comment by AndreiMisiukevich** [comment]
Hey folks @PureWeen @hartez 
Do you have any idea of how to fix this issue? I'm pretty sure we should resolve it in iOS CollectionViewRenderer

**Comment by 9mrcookie9** [comment]
Hello,
I can confirm that the problem occurs on emulators and physical devices. (iOS)

It seems to me that the problem here is how Expander is reused by CollectionView. Try to add a (Bindable property) "expanded" to your data model, and bind it to "IsExpanded" field with two way binding.
Probably fields like height are also copied. So there is no problem with Expander control itself.

I personally got to it after changing the control to ListView and experimenting with element recycling modes.

**Comment by tommy-elliott-awh** [comment]
As stated in previous comments, Expander works fine with Scroll + Stack.  Here I give more specifics about how to apply that as a workaround, when not requiring CollectionView or ListView specific functionality.  Change the container from
```
<CollectionView ItemsSource="{Binding MyItems}">
    <CollectionView.ItemTemplate>
        <!--Expander and possibly other content-->
    </CollectionView.ItemTemplate>
</CollectionView>
```
to
```
<ScrollView>
    <StackLayout Spacing="0" BindableLayout.ItemsSource="{Binding MyItems}">
        <BindableLayout.ItemTemplate>
            <!--Expander and possibly other content-->
        </BindableLayout.ItemTemplate>
    </StackLayout>
</ScrollView>
```
This workaround resolves all expand/collapse issues on iOS and Android for me in this scenario (and also on Xamarin Forms 5.0), including the expand+scroll or collapse+scroll issue where when under CollectionView expand/collapse of one item would incorrectly expand/collapse other items further down/up the list once or twice each screen's worth of vertical space because of CollectionView cell reuse.

**Comment by Cfun1** [comment]
This issue needs to have iOS label, right ?

**Comment by sachinkanadia** [comment]
@Cfun1 Yes I should think so

@tommy-elliott-awh Using bindable layout with expander view does not work well on Android. Scrolling is glitchy and not smooth.

My workaround for this issue has been to implement a Bindable Layout for iOS and Collectionview for Android. The user will experience the same UX based on this implementation however the underlying code becomes platform specific and difficult to maintain. Works for the time being though.

This issue remains to be resolved.

**Comment by jamiekeefer** [comment]
@tommy-elliott-awh suggestion worked for the time being. Did not experience the scrolling glitches mentioned by @sachinkanadia however the page does appear to take longer to load in both android and iOS. This has pluses and minuses in that with the page and all the content loaded, I find the subsequent navigation smoother than having to wait for lazy loaded content - which can cause some unresponsive scrolling UX.

**Comment by jfversluis** [comment]
We've added to the [Docs](https://docs.microsoft.com/xamarin/community-toolkit/views/expander): Expander and CollectionView/ListView should not be used together. I'd love to have a solution for this but it’s one of those controls that don’t exist on the native platform. Because of that you will be fighting the OS on getting it right. Closing this one since there won't be any solution on the short-term unless someone contributes it.
